### PR TITLE
Idx list of all images by date #422

### PR DESCRIPTION
**Pull request by miyunari**
part of #422

### CODE REVIEW

File: src/cloudimagedirectory/transform/transform.py [hunk 1]
```diff
@@ -41,6 +42,62 @@ def run(self, data):
...
+    """Transform raw AWS data."""
+
+    def run(self, data):
+        """Transform the raw data."""
+        # Verify that the data is from AWS.
+        entries = [x for x in data if not x.is_raw()]
+
+        # Sort the list of data by date
+        entries.sort(
+            key=lambda x: datetime.strptime(
```
Comment (major) [inline src/cloudimagedirectory/transform/transform.py:55]: Ooh, lambda time! :nerd_face:

File: src/cloudimagedirectory/transform/transform.py [hunk 2]
```diff
@@ -41,6 +42,62 @@ def run(self, data):
...
+                "".join(x.content["date"].split("T")[0]), "%Y-%m-%d"
+            ),
+            reverse=True,
+        )
+
+        list = []
+        for e in entries:
```
Comment (major) [inline src/cloudimagedirectory/transform/transform.py:62]: In Python, single letter variables are avoided outside of small structures like lambdas or dictionary comprehension.

`for entry in entries` would be fine here.

File: src/cloudimagedirectory/transform/transform.py [hunk 3]
```diff
@@ -41,6 +42,62 @@ def run(self, data):
...
+        for i, item in enumerate(list, 1):
+            chunk.append(item)
+            if len(list) == i or i % chunk_size == 0:
+                chunked_list.append(chunk)
+                chunk = []
+
+        results = []
+        for page in range(0, len(chunked_list)):
+            data_entry = connection.DataEntry(
+                f"idx/list/sort-by-date/{page}", chunked_list[page]
```
Comment (major) [inline src/cloudimagedirectory/transform/transform.py:94]: Nice use of f-strings! :clap:

**Review by major — APPROVED** [review]
A few minor things. This looks okay otherwise if you want to merge it as-is.

File: src/cloudimagedirectory/transform/transform.py (hunk 2 above)
Comment (miyunari) [reply src/cloudimagedirectory/transform/transform.py:62]: aaah, right - will change it :smile:

**Review by FKolwa — APPROVED** [review]

**Comment by FKolwa** [comment]
Great work! Thanks for getting this done ;)

**Pull request merged**